app/src/app.py

```diff
@@ -1,7 +1,51 @@
-from flask import Flask, render_template
-
+from flask import Flask, render_template, request, redirect, flash, url_for
+from werkzeug.security import generate_password_hash
 app = Flask(__name__)
 
 @app.route('/')
 def index():
     return render_template('index.html')
+
+@app.route("/register", methods=["GET", "POST"])
+def register():
+    """Register user"""
+
+    if request.method == "POST":
+
+        #Access the data submitted to form
+        email = request.form.get("email")
+        username = request.form.get("username")
+        password = request.form.get("password")
+
+
+        #Check if the inputs are valid
+        user_email = User.query.filter_by(email=email).first()
+        if user_email:
+            flash("This email already exists.")
+            return redirect(url_for("register"))
```

Comment:
It may be a good idea to think of a different way than to redirect the user in the case of an unhappy path. Some things we could do is display an error message instead. This isn’t a huge deal, but could be a feature improvement